tests/xharness/Harness.cs

```diff
@@ -31,16 +31,42 @@ public interface IHarness {
 		Version XcodeVersion { get; }
 		Task<ProcessExecutionResult> ExecuteXcodeCommandAsync (string executable, IList<string> args, ILog log, TimeSpan timeout);
 	}
+	public class HarnessConfiguration {
+		public bool AutoConf { get; set; }
+		public string Configuration { get; set; } = "Debug";
+		public bool DryRun { get; set; }
+		public Dictionary<string, string> EnvironmentVariables { get; set; } = new Dictionary<string, string> ();
+		public bool? IncludeSystemPermissionTests { get; set; }
+		public List<iOSTestProject> IOSTestProjects { get; set; } = new List<iOSTestProject> ();
+		public string JenkinsConfiguration { get; set; }
+		public HashSet<string> Labels { get; set; } = new HashSet<string> ();
+		public string LogDirectory { get; set; } = Environment.CurrentDirectory;
+		public string LogFile { get; set; }
+		public bool Mac { get; set; }
+		public string MarkdownSummaryPath { get; set; }
+		public string PeriodicCommand { get; set; }
+		public string PeriodicCommandArguments { get; set; }
+		public TimeSpan PeriodicCommandInterval { get; set; }
+		public string RootDirectory { get; set; }
+		public string SdkRoot { get; set; }
+		public AppRunnerTarget Target { get; set; }
+		public double TimeoutInMinutes { get; set; } = 15;
+		public bool UseSystem { get; set; }
+		public int Verbosity { get; set; }
+		public string WatchOSAppTemplate { get; set; }
+		public string WatchOSContainerTemplate { get; set; }
+		public XmlResultJargon XmlJargon { get; set; } = XmlResultJargon.NUnitV3;
+	}
 
 	public class Harness : IHarness
 	{
-		public HarnessAction Action { get; set; }
-		public int Verbosity { get; set; }
+		public HarnessAction Action { get; }
+		public int Verbosity { get; }
 		public ILog HarnessLog { get; set; }
-		public bool UseSystem { get; set; } // if the system XI/XM should be used, or the locally build XI/XM.
-		public HashSet<string> Labels { get; } = new HashSet<string> ();
-		public XmlResultJargon XmlJargon { get; set; } = XmlResultJargon.NUnitV3;
-		public IProcessManager ProcessManager { get; set; } = new ProcessManager ();
+		public bool UseSystem { get; } // if the system XI/XM should be used, or the locally build XI/XM.
```

Comment:
`UseSystem` is a vague name. Is there a reason why it couldn't be named `UseSystemXamarinIOSMac` or with the opposite connotation `UseLocalXamarinIOSMac`?

Reply:
I would not rename fields as part of this refactoring, if possible

Reply:
Although, this wouldn't hurt

Reply:
Done